### DISCUSSION TITLE
Can I use this script along with Chris Titus' script

### DISCUSSION THREAD

**Discussion by catsz** [simeononsecurity/Windows-Optimize-Debloat#11] [Q&A] [answered]
I saw your comment on Chris Titus' website showing off this script (https://christitus.com/windows-11-perfect-install/). I am wondering if I could run both scripts or if your script quite literally does the same thing but more. I mainly want to run his script to easily remove Cortana and to have more optimization if possible.

I would also like to know exactly how the update functionality is restricted.

**Comment by simeononsecurity** [comment, marked as answer]
> I saw your comment on Chris Titus' website showing off this script (https://christitus.com/windows-11-perfect-install/). I am wondering if I could run both scripts or if your script quite literally does the same thing but more. I mainly want to run his script to easily remove Cortana and to have more optimization if possible.
> 
> I would also like to know exactly how the update functionality is restricted.

It isn't suggested to remove cortana. It is backed into windows very hard. removing it breaks more than I can possibly explain in one message. My script disables it and removes it as far as possible without breaking anything other than cortana.

Update functionality on this script restricts sharing updates with other computers on the internet and on your local network. It only allows updates to be downloaded from straight from WSUS (if you're in an enterprise) or otherwise direct from microsoft. Lastly it also disables driver updates through windows. As it is best practice to download the drivers directly from the manufacturers. Drivers on windows updates lag behind months and often come with less compatibility and more issues than direct from the manufacturer. 
Chris's script is written by a linux extremist and a community of people that don't have a great understanding of windows. It quite literally doesn't implement things to windows best practices and breaks somethings beyond the ability to restore. That being said, it does do some things better than mine, ex the gui is better. Also mine isn't perfect either, no privacy script really can be. But I have to say I'm not a fan.

But to answer your question, I have not tested running both his project and mine. Mine makes over a million configuration changes on a windows system. Testing every possible combination of our script with others is just unreasonable. It's ideal and preferred for you to run our script only on a clean install of windows. However, there is no reason why it couldn't work. Just obviously we can't guarantee it won't break anything by combining the two. 

As always take a backup, snapshot, or at least a restore point before running any script from the internet.
